utilities/table_properties_collectors/compact_on_deletion_collector.cc

```diff
@@ -118,5 +128,34 @@ NewCompactOnDeletionCollectorFactory(size_t sliding_window_size,
       new CompactOnDeletionCollectorFactory(sliding_window_size,
                                             deletion_trigger, deletion_ratio));
 }
-}  // namespace ROCKSDB_NAMESPACE
+namespace {
+static int RegisterTablePropertiesCollectorFactories(
+    ObjectLibrary& library, const std::string& /*arg*/) {
+  library.Register<TablePropertiesCollectorFactory>(
+      CompactOnDeletionCollectorFactory::kClassName(),
+      [](const std::string& /*uri*/,
+         std::unique_ptr<TablePropertiesCollectorFactory>* guard,
+         std::string* /* errmsg */) {
+        guard->reset(new CompactOnDeletionCollectorFactory(0, 0, 0));
```

Comment:
Is "0, 0, 0" a correct default initialization argument?

Reply:
> Is "0, 0, 0" a correct default initialization argument?

I missed putting in OptionType parameters for the CompactOnDeletionCollectorFactory earlier -- since corrected.

As far as defaults, the first and last 0 (window size, deletion ratio) should effectively disable this collector.  The middle argument (deletion trigger) really be set to a meaningful value for the user -- we could make a guess as to a different default but I do not know what that should reasonably be.

Reply:
Can we avoid this default construction here? I'm not comfortable with the "0,0,0" arguments. Or maybe, at least, you should add the comments say  by default, the collector is disabled since we give 0 window size.

Reply:
Added comment to the code.